### ISSUE TITLE
Closing groups results in crash

### ISSUE THREAD

**Issue by zacharycarter** [vurtun/nuklear#350]
I'm not sure if this is a bug or simply something I'm not handling correctly but what I've found is that I can create a group with a window flag of NK_WINDOW_CLOSABLE and there doesn't seem to be a a way to handle the close event for the group.

The crash / bug is pretty easy to reproduce, just add this flag to any group begin statement and then try to find a way to handle the close event.

Doing an if check on the nk_group_begin and then trying to conditionally render the group when this returns false, does not work. It seems to always return true, even if the close button on the group is clicked.

If this is a known issue, please disregard.

Thanks!

-Zachary Carter

**Comment by vurtun** [comment]
Groups in nuklear have no closable implementation at the moment (I forgot to add an assert to check for it). I don't really now if groups should be closable. In theory it is possible. In addition to storing groups scrollbar x- and y-offset you could also store the state (active,inactive). Is there any specific use case? You can just manage the state outside and not call `nk_group_begin`. Of course this does not preserve scrollbar offsets. But even they could be managed outside. Another question is how would you reopen a previously closed group. So some additional functions would be required to make it work.

**Comment by zacharycarter** [comment]
Hmm these are good points I didn't think about. Perhaps the best route would just be the assert you mentioned. I agree with you, it doesn't really make sense for them to be closable since technically they are layouts and not widgets right?

What I'm trying to do specifically is create a window nested inside of another window, like this - 
http://imgur.com/a/LymN9

Originally I figured a closable group would be the best option for the inner window, until I ran into that bug.

I switched to instead drawing two windows, one for the menu and then one for the rest of the content in the screenshot I shared. I call nk_begin / nk_end for the menu and then I call nk_begin / nk_end again for the second window. I'm running into a new problem now though - when I have the second window open and I focus on the menu, I'm unable to refocus on the second window.

I'm guessing all this is because I'm not using popups? I believe that's the widget I should be using here...

Thanks for the reply / help as usual!
